### PR TITLE
feat: add cost-analyst and cost-reviewer plugin agents (#433)

### LINKED ISSUE #433: Plugin: Extract agents into standalone .agent.md files
### Description

Extract the agent definitions currently in `AGENTS.md` into individual `.agent.md` files in a dedicated `agents/` directory. Each agent gets proper YAML frontmatter with tool scoping, enabling per-agent dispatch, parallel execution, and context isolation.

### Dependencies
- Depends on #432 (`plugin.json` must declare `agents` path first)

### ⚠️ Design decisions requiring sign-off before implementation
The following decisions must be brainstormed and agreed upon before any code changes:
1. **Agent file format**: `.agent.md` with YAML frontmatter is the research consensus — confirm this is the right format for our target platforms
2. **Tool scoping per agent**: What tools does each agent get? Research suggests least-privilege — each agent only accesses what it needs. Need to define the tool set per agent:
   - `pricing-investigator` — shell (pricing scripts), read, search
   - `compliance-reviewer` — read (templates/schema), search
   - `service-reference` (orchestrator) — all tools + agent dispatch
   - `service-ref-pr-reviewer` — read, search, diff/review tools
3. **AGENTS.md future**: Does it become a lightweight index pointing to agent files, or do we remove it entirely?
4. **New agents**: Do we add any new agents beyond the existing 4? (e.g., `explore-pricing` for interactive API exploration)
5. **Agent dispatch model**: Inference-based (AI picks the right agent from descriptions) vs explicit (command triggers specific agent) vs both?

### Research Context

**Agent file format** (consensus across Opus, GPT, Gemini):

Agents are Markdown files with YAML frontmatter defining role, tools, and behavior:

```markdown
---
name: pricing-investigator
description: >
  Investigates the Azure Retail Prices API for a given service,
  cataloging meters, SKUs, products, and billing boundaries.
tools:
  - read
  - search
  - shell
user-invocable: true
metadata:
  role: investigator
  domain: azure-pricing
---

# Pricing Investigator

You are an Azure pricing investigation specialist...

## Workflow
1. Use explore script to discover all meters
2. Categorize by billing dimension
3. Identify pricing tiers and variations
4. Document edge cases
5. Return structured findings
```

**Key frontmatter fields** (from GitHub Copilot CLI agent spec):
| Field | Type | Description |
|---|---|---|
| `name` | string | Display name |
| `description` | string (required) | What the agent does — used for inference matching |
| `tools` | string[] | Tools the agent can use |
| `disable-model-invocation` | bool | Require manual selection only |
| `user-invocable` | bool | Whether users can launch directly |
| `mcp-servers` | object | Additional MCP tool definitions |
| `metadata` | object | Arbitrary annotations |

**Agent storage locations** (platforms):
- Copilot CLI: `agents/` in plugin, `.github/agents/` in repo
- Claude Code: `.claude/agents/` directory
- Factory AI: `droids/` directory

**Current agents in AGENTS.md:**
- `pricing-investigator` — Investigates Azure Retail Prices API
- `compliance-reviewer` — Validates service reference files against rules/templates
- `service-reference` — Orchestrator dispatching sub-agents
- `service-ref-pr-reviewer` — Reviews PRs for service reference changes

### Scope

**Discovery / Analysis**
- Finalize agent file format and YAML frontmatter schema
- Determine tool scoping per agent (least privilege)
- Research how target platforms discover and dispatch plugin agents
- Determine what happens to `AGENTS.md`

**Implementation** (only after design decisions above are signed off)
- Create `agents/` directory
- Extract each agent into its own file
- Add YAML frontmatter with tool restrictions, description, and metadata
- Refactor `AGENTS.md` to be a lightweight index/overview
- Update `plugin.json` to reference `agents/` directory

**Testing**
- Validate agent files are discoverable by target platforms
- Validate per-agent dispatch works (agent loads only its own context)
- Validate context isolation (sub-agent work doesn't pollute main context)
- Confirm existing AGENTS.md-based workflows still function during transition

### Expected Results
- Each agent is independently loadable with its own context window
- Parallel dispatch is possible (multiple agents investigate different services simultaneously)
- Context isolation reduces main context consumption for complex architectures
- Directly addresses **H2** (parallel sub-agent dispatch) and **H3** (context isolation reduces hallucinations)

### Parent Issue
Part of #431

### PR DESCRIPTION
**Pull request by ahmadabdalla**
## Summary

Closes #433 (child of #431).

Adds two standalone plugin agent files to `agents/` — **cost-analyst** and **cost-reviewer** — for distribution with the plugin package. These agents provide structured workflows for Azure cost estimation and review, usable on both GitHub Copilot CLI and Claude Code.

### New files
- **`agents/cost-analyst.agent.md`** (148 lines) — Primary user-facing agent for cost assessments. Orchestrates the full SKILL.md workflow: intake → service grouping → API queries → arithmetic → presentation. Includes review gate via cost-reviewer dispatch.
- **`agents/cost-reviewer.agent.md`** (120 lines) — Quality-gate agent that validates cost assessments for arithmetic accuracy, completeness, and consistency. Invoked by cost-analyst or directly by users.
- **`docs/plugin-agents.md`** (231 lines) — Plugin agent documentation covering cross-platform compatibility, discovery paths, consumer runtime considerations, and development guidelines.

### Modified files
- **`docs/ops/custom-agents.md`** — Added cross-references to distinguish CI agents (`.github/agents/`) from plugin agents (`agents/`).
- **`skills/azure-cost-calculator/SKILL.md`** — Minor reference update.
- **`.github/workflows/create-release.yml`** — Unrelated workflow fix (carried from dev).

## Design decisions

| # | Decision | Choice |
|---|----------|--------|
| DD1 | Agent scope | Two consumer-focused agents (cost-analyst, cost-reviewer) — distinct from the 4 CI agents in `.github/agents/` |
| DD2 | File format | YAML frontmatter with `name` + `description` — cross-compatible with Copilot CLI (derives ID from filename) and Claude Code (reads frontmatter) |
| DD3 | Extension | `.agent.md` — matches both platform glob patterns (`*.md` for Claude Code, `*.agent.md` convention for Copilot CLI) |
| DD4 | No ID collisions | Plugin agents use `cost-analyst`/`cost-reviewer`; CI agents use `pricing-investigator`/`compliance-reviewer`/`service-reference`/`service-ref-pr-reviewer` |

## Testing — 3×3×3 comparison (9 runs)

Ran the same complex architecture (Trade Settlement & AML Event Processing — 15+ services, AUD, 1-Year RI) across three groups:

| Group | Runs | Avg Total | Std Dev | Avg Host Tokens | Platform |
|-------|------|-----------|---------|-----------------|----------|
| Copilot CLI + Agent | 1, 2, 3 | AUD $3,024.47 | $0.13 | 108.3k (54%) | Copilot CLI |
| Copilot CLI Raw | 4, 5, 6 | AUD $2,881.66 | $10.81 | 96.3k (48%) | Copilot CLI |
| Claude Code + Agent | 7, 8, 9 | AUD $2,946.50 | $93.55 | 32k (16%) | Claude Code |

### Key findings

1. **Cross-platform compatibility** ✅ — Agents discovered and functional on both Copilot CLI and Claude Code
2. **Dispatch model differs by platform** — Copilot CLI runs estimation in-host with agent spec as context; Claude Code dispatches cost-analyst as a true autonomous sub-agent (56-109 tool calls)
3. **Host token efficiency** — Claude Code: 32k host tokens (sub-agent uses ~99-113k internally in separate context); Copilot CLI: 96-114k in shared context
4. **Sentinel billing non-deterministic** — 5 distinct interpretations across 9 runs ($155 spread). This is a service reference gap, not an agent issue — tracked separately (#446)
5. **cost-reviewer dispatch** — Triggered on Copilot CLI (2/3 agent runs) but never on Claude Code (0/3) due to single-level sub-agent dispatch
6. **VNet architecture conflict** — One Claude Code run (Run 9, 109 tool calls) surfaced a legitimate Functions Consumption + Private Endpoints incompatibility that no other run caught (#449)

## Not included

- Scratchpad files (`scratchpad/433-*.md`) — local research and test results, not committed
- Pre-computation orchestration changes — tested in Run 3, validated, but reverted pending further iteration
- T4-T9 test scenarios — designed but not yet executed (clarification flow, iteration, comparative, reviewer edge cases)

## Summary by CodeRabbit

* **New Features**
  * Added two agents for Azure cost estimation and validation workflows: a Cost Analyst (estimation/orchestration) and a Cost Reviewer (structured validation and reporting)

* **Documentation**
  * Added comprehensive Plugin Agents guidance covering agent types, loading, invocation, and runtime considerations
  * Reorganized Custom Agents docs with improved structure, terminology, and a new Platform Documentation section

### CODE REVIEW

File: docs/plugin-agents.md [hunk 1]
```diff
@@ -0,0 +1,231 @@
...
+
+- `cost-analyst`: shell execution (pricing scripts), file reading, file searching, agent dispatch (to cost-reviewer)
+- `cost-reviewer`: file reading, file searching only (no shell, no edit — principle of least privilege)
+
+### VS Code parser constraints
+
+The VS Code agent parser is strict about formatting:
+
+- **No YAML folded scalars** (`>` or `|`) — causes "Unexpected indentation" errors. Use single-line quoted strings instead.
+- **Supported VS Code attributes**: `agents`, `argument-hint`, `description`, `disable-model-invocation`, `handoffs`, `model`, `name`, `target`, `tools`, `user-invokable`. Unknown fields produce warnings in the editor but are harmless at runtime.
```
[the remark this answers is not in the document]
Comment (ahmadabdalla) [reply docs/plugin-agents.md]: Fixed in d2aa31b — corrected `user-invokable` → `user-invocable` to match the actual VS Code attribute name.

File: docs/plugin-agents.md [hunk 2]
```diff
@@ -0,0 +1,230 @@
...
+
+- [Custom agents](https://docs.github.com/en/copilot/how-tos/use-copilot-agents/coding-agent/create-custom-agents)
+- [Custom agents configuration](https://docs.github.com/en/copilot/reference/custom-agents-configuration)
+
+---
+
+## Related
+
+- [CI/ops custom agents](ops/custom-agents.md) — repository governance agents in `.github/agents/`
+- [Plugin manifest](../skills/azure-cost-calculator/SKILL.md) — skill entry point
```
[the remark this answers is not in the document]
Comment (ahmadabdalla) [reply docs/plugin-agents.md]: Fixed in d2aa31b — split the link into two entries: "Skill entry point" pointing to SKILL.md, and "Plugin manifest" pointing to `.claude-plugin/plugin.json`.
